lib/internal/crypto/random.js

```diff
@@ -164,36 +171,59 @@ function randomInt(min, max, callback) {
   const excess = RAND_MAX % range;
   const randLimit = RAND_MAX - excess;
 
-  if (isSync) {
-    // Sync API
-    while (true) {
-      const x = randomBytes(6).readUIntBE(0, 6);
-      // If x > (maxVal - (maxVal % range)), we will get "modulo bias"
-      if (x > randLimit) {
-        // Try again
-        continue;
-      }
+  // If we don't have a callback, or if there is still data in the cache, we can
+  // do this synchronously, which is super fast.
+  while (isSync || (randomCacheOffset < randomCache.length)) {
+    if (randomCacheOffset === randomCache.length) {
+      // This might block the thread for a bit, but we are in sync mode.
+      randomFillSync(randomCache);
+      randomCacheOffset = 0;
+    }
+
+    const x = randomCache.readUIntBE(randomCacheOffset, 6);
+    randomCacheOffset += 6;
+
+    // If x <= (maxVal - (maxVal % range)), we have avoided "modulo bias".
+    if (x <= randLimit) {
```

Comment:
Shouldn't this check be `if (x < randLimit)` in order to avoid modulo bias? Including `randLimit` adds +1 "slot" (event) for `0` outcome in terms of probability.

This also applies to the original code which had the following check:
```js
      if (x > randLimit) {
        // Try again
        continue;
      }
```

Reply:
@puzpuzpuz I separated the changes, see https://github.com/nodejs/node/pull/36894. I added you as a co-author on that commit, but do let me know if you have any objections to that.

Reply:
Thanks for addressing the issue. Going to review #36894.